Can't apt update: NO_PUBKEY 04EE7237B7D453EC NO_PUBKEY 648ACFD622F3D138

After following these steps on a fresh osmc install on a raspberry pi zero, I can't install wireguard on the last step because `apt update` fails:

```
$ sudo apt update
Hit:1 http://mirrordirector.raspbian.org/raspbian buster InRelease
Get:3 http://deb.debian.org/debian unstable InRelease [153 kB]                    
Hit:2 http://ftp.fau.de/osmc/osmc/apt buster InRelease                            
Err:3 http://deb.debian.org/debian unstable InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 04EE7237B7D453EC NO_PUBKEY 648ACFD622F3D138
Reading package lists... Done
W: GPG error: http://deb.debian.org/debian unstable InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 04EE7237B7D453EC NO_PUBKEY 648ACFD622F3D138
E: The repository 'http://deb.debian.org/debian unstable InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```

OS version: 

```
$ cat /etc/debian_version
10.7
$ cat /etc/os-release
PRETTY_NAME="Open Source Media Center"
NAME="OSMC"
VERSION="November 2020"
VERSION_ID="2020.11-1"
$ cat uname -r
4.19.122-2-osmc
```

I'll comment if I find a solution.

I was able to get around the above issue by doing:

```
sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 04EE7237B7D453EC
sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 648ACFD622F3D138
```

I can know update but installing wireguard fails with the following output.

```
$ sudo apt install wireguard-dkms wireguard-tools wireguard
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  dkms
Suggested packages:
  python3-apport menu openresolv | resolvconf
Recommended packages:
  fakeroot linux-headers-686-pae | linux-headers-amd64 | linux-headers-generic | linux-headers
The following NEW packages will be installed:
  dkms wireguard wireguard-dkms wireguard-tools
0 upgraded, 4 newly installed, 0 to remove and 0 not upgraded.
Need to get 414 kB of archives.
After this operation, 2,339 kB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://deb.debian.org/debian unstable/main armhf wireguard-dkms all 1.0.20201221-1 [254 kB]
Get:3 http://deb.debian.org/debian unstable/main armhf wireguard-tools armhf 1.0.20200827-1 [77.8 kB]
Get:4 http://deb.debian.org/debian unstable/main armhf wireguard all 1.0.20200827-1 [8,104 B]
Get:2 http://mirror.umd.edu/raspbian/raspbian buster/main armhf dkms all 2.6.1-4 [74.4 kB]
Fetched 414 kB in 1s (300 kB/s)                   
Selecting previously unselected package dkms.
(Reading database ... 50380 files and directories currently installed.)
Preparing to unpack .../archives/dkms_2.6.1-4_all.deb ...
Unpacking dkms (2.6.1-4) ...
Selecting previously unselected package wireguard-dkms.
Preparing to unpack .../wireguard-dkms_1.0.20201221-1_all.deb ...
Unpacking wireguard-dkms (1.0.20201221-1) ...
Selecting previously unselected package wireguard-tools.
Preparing to unpack .../wireguard-tools_1.0.20200827-1_armhf.deb ...
Unpacking wireguard-tools (1.0.20200827-1) ...
Selecting previously unselected package wireguard.
Preparing to unpack .../wireguard_1.0.20200827-1_all.deb ...
Unpacking wireguard (1.0.20200827-1) ...
Setting up dkms (2.6.1-4) ...
Setting up wireguard-dkms (1.0.20201221-1) ...
Loading new wireguard-1.0.20201221 DKMS files...
Building for 4.19.122-2-osmc
Building initial module for 4.19.122-2-osmc
Error! Bad return status for module build on kernel: 4.19.122-2-osmc (armv6l)
Consult /var/lib/dkms/wireguard/1.0.20201221/build/make.log for more information.
dpkg: error processing package wireguard-dkms (--configure):
 installed wireguard-dkms package post-installation script subprocess returned error exit status 10
Setting up wireguard-tools (1.0.20200827-1) ...
wg-quick.target is a disabled or a static unit, not starting it.
dpkg: dependency problems prevent configuration of wireguard:
 wireguard depends on wireguard-modules (>= 0.0.20191219) | wireguard-dkms (>= 0.0.20200121-2); however:
  Package wireguard-modules is not installed.
  Package wireguard-dkms is not configured yet.

dpkg: error processing package wireguard (--configure):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 wireguard-dkms
 wireguard
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

The make log shows a segmentation fault.

```
$ cat /var/lib/dkms/wireguard/1.0.20201221/build/make.log
DKMS make.log for wireguard-1.0.20201221 for kernel 4.19.122-2-osmc (armv6l)
Thu 31 Dec 2020 01:23:53 AM EST
make: Entering directory '/usr/src/rbp2-headers-4.19.122-2-osmc'
  CC [M]  /var/lib/dkms/wireguard/1.0.20201221/build/main.o
Segmentation fault
make[1]: *** [scripts/Makefile.build:304: /var/lib/dkms/wireguard/1.0.20201221/build/main.o] Error 139
make: *** [Makefile:1522: _module_/var/lib/dkms/wireguard/1.0.20201221/build] Error 2
make: Leaving directory '/usr/src/rbp2-headers-4.19.122-2-osmc'
```

The error persists after a reboot. Not sure how to proceed.

--- a/README.md
+++ b/README.md
@@ -2,6 +2,8 @@
 
 ## 1. Prepare the system
 
+Note: Raspberry Pi 1 & Zero must compile WireGuard manually and may skip this step.
+
 ```console
 osmc@osmc:~$ sudo apt update
 osmc@osmc:~$ sudo apt upgrade
@@ -32,6 +34,20 @@ osmc@osmc:/usr/src$ sudo ln -s /usr/src/rbp2-headers-`uname -r` /lib/modules/`un
 osmc@osmc:/usr/src$ sudo rm rbp2-source-`uname -r`.tar.bz2
 ```
 
+For Raspberry Pi 1 & Zero, use `rbp1-headers` and `rbp1-source`.
+
+
+```console
+osmc@osmc:~$ sudo apt-get install rbp1-headers-$(uname -r) rbp1-source-$(uname -r) libmnl-dev libelf-dev build-essential git pkg-config
+
+osmc@osmc:~$ cd /usr/src/
+osmc@osmc:/usr/src$ sudo rm -r rbp1-headers-`uname -r`/include/linux/*
+osmc@osmc:/usr/src$ sudo tar -xvf rbp1-source-`uname -r`.tar.bz2 rbp1-source-`uname -r`/include/linux
+osmc@osmc:/usr/src$ sudo cp -ar ./rbp1-source-`uname -r`/include/linux/* ./rbp1-headers-`uname -r`/include/linux/
+osmc@osmc:/usr/src$ sudo ln -s /usr/src/rbp1-headers-`uname -r` /lib/modules/`uname -r`/build
+osmc@osmc:/usr/src$ sudo rm rbp1-source-`uname -r`.tar.bz2
+```
+
 ## 3. WireGuard installation
 
 First installation:
@@ -45,6 +61,18 @@ After updating kernel, repeat `2.` and:
 
 ```console
 osmc@osmc:~$ sudo dpkg-reconfigure wireguard-dkms
+```
+
+For Raspberry Pi 1 & Zero, WireGuard must be compiled manually.
+
+```console
+osmc@osmc:~$ cd ~
+osmc@osmc:~$ git clone https://git.zx2c4.com/wireguard-linux-compat
+osmc@osmc:~$ git clone https://git.zx2c4.com/wireguard-tools
+osmc@osmc:~$ make -C wireguard-linux-compat/src -j$(nproc)
+osmc@osmc:~$ sudo make -C wireguard-linux-compat/src install
+osmc@osmc:~$ make -C wireguard-tools/src -j$(nproc)
+osmc@osmc:~$ sudo make -C wireguard-tools/src install
 ```
 
 ## Configure WireGuard

